solidity/ecdsa/contracts/libraries/EcdsaDkg.sol

```diff
@@ -71,6 +75,10 @@ library EcdsaDkg {
         bytes32 submittedResultHash;
         // Block number from the moment of the DKG result submission.
         uint256 submittedResultBlock;
+        // Reserved storage space in case we need to add more variables.
+        // See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+        // slither-disable-next-line unused-state
+        uint256[11] __gap;
```

Comment:
Continuing the thread about gaps: Everything in `Data`, including`Parameters`, take 12 storage slots. I think we should have `uint256[38] __gap;` but only at the end of `Data`. I would remove the gap in `Parameters` and in case we need to add more parameters, I would solve it by `ParametersV2` struct. I think this approach is more flexible because we have the freedom to use our 38 storage slots in any way we want.

Reply:
e74e17d368484b95abbe2c2d4ae4b6a57d97a2b7

Reply:
Let's not include what's inside the Parameters structure to calculate a gap for `Data`, same as you calculated a gap for `EcdsaAuthorization` in https://github.com/keep-network/keep-core/pull/2926#discussion_r862463999


So here we would have 1 slot for Parameters and 8 for other elements, so the gap would be `41`.

Reply:
I did include what's in `Parameters` structure to calculate a gap for `EcdsaAuthorization`. `EcdsaAuthorization.Parameters` occupy one storage slot because storage is tightly packed inside a structure. We have 224 bits there, so it's one EVM word. We even have a space for another `uint32` there.

I do think it makes sense to include what's inside `Parameters`. This way, each library has 50 storage slots consistently.

Reply:
> I did include what's in Parameters structure to calculate a gap for EcdsaAuthorization. EcdsaAuthorization.Parameters occupy one storage slot because storage is tightly packed inside a structure. We have 224 bits there, so it's one EVM word. We even have a space for another uint32 there.

Wow, you're even more clever than I thought! 😉 

But.. I don't like getting that much into detail to calculate the gap. Should we analyze the type of each element in the struct? What about [mappings and dynamic arrays](https://docs.soliditylang.org/en/v0.8.13/internals/layout_in_storage.html#mappings-and-dynamic-arrays) - should they be included when calculating the gap or not? IMHO it looks like we're overthinking it.

I can agree to include elements of `Parameters` struct in the gap in `Data`, but each element with the same wage, no matter how they are packed in slots, so `EcdsaAuthorization.Parameters` will subtract `3` from the gap.

Reply:
And we should subtract mappings from the gap as well. Each mapping is `- 1`. Just to simplify the operation of calculating the gap.

Reply:
> But.. I don't like getting that much into detail to calculate the gap. Should we analyze the type of each element in the struct?

Yes, this is giving us gas savings when reading and writing. That's why we do not use `uint256` everywhere but go down to `uint64` and `uint32`. We not only need to analyze the type of each element but also how they are ordered so that tightly packing is possible.

> What about [mappings and dynamic arrays](https://docs.soliditylang.org/en/v0.8.13/internals/layout_in_storage.html#mappings-and-dynamic-arrays) - should they be included when calculating the gap or not?

> And we should subtract mappings from the gap as well. Each mapping is - 1. Just to simplify the operation of calculating the gap.

We need to assume each mapping and dynamic array (including a string) occupies one storage slot. This how we calculate the gap in `EcdsaAuthorization`. 

Examples from OpenZeppelin:
- https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/v4.5.2/contracts/finance/PaymentSplitterUpgradeable.sol
- https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/v4.5.2/contracts/token/ERC777/ERC777Upgradeable.sol
- https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/v4.5.2/contracts/token/ERC20/ERC20Upgradeable.sol
- https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/v4.5.2/contracts/finance/PaymentSplitterUpgradeable.sol

Reply:
> > But.. I don't like getting that much into detail to calculate the gap. Should we analyze the type of each element in the struct?
> 
> Yes, this is giving us gas savings when reading and writing. That's why we do not use `uint256` everywhere but go down to `uint64` and `uint32`. We not only need to analyze the type of each element but also how they are ordered so that tightly packing is possible.

I know that typing and ordering is crucial for storage slots consumption. What I mean is that we don't need to go into such details when calculating the gap. We could simplify it and assume that each element subtracts 1 from the gap (50). As we did for mappings.

Reply:
A gap is a collection of 256-bit elements, so it's a collection of EVM words. 

Gap has 50 elements by default, so 50 EVM 256-bit words. 

If we subtract always one element, no matter how much storage space is actually occupied, for `Parameters{uint64, uint64, uint128}`, we would end up with 1 EVM word used by the storage and 47 EVM words in a gap. 1 + 47 != 50, so this is inconsistent. 


> As we did for mappings.

Mapping and array do occupy one storage slot, that's why we subtracted 1 EVM word (or 256 bits). True, those are dynamic data types, and elements are written into random slots. But dynamic data types will always be such special cases and that one storage slot we subtracted is actually occupied. Worth noting, where actually data lands is computed based on the one slot occupied by the element in the storage we are discussing.

Reply:
You won 😸 80895aab2cbac1f8df04b209f7ce64ba01cd93db